(common): Constructs are incompatible with NodeJS 25

### Describe the bug

When trying to install the package in a repo using NodeJS 25.X for the engine, I get the error

```
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with "@cdklabs/generative-ai-cdk-constructs@0.1.316(aws-cdk-lib@2.250.0(constructs@10.6.0))(constructs@10.6.0)".

Expected version: >= 20.x <= 24.x
Got: v25.9.0

This is happening because the package's manifest has an engines.node field specified.
To fix this issue, install the required Node version.
```

### Expected Behavior

I expect the constructs to support NodeJS 25.X

### Current Behavior

Install fails

### Reproduction Steps

1. Create a repo that uses NodeJS 25.x as the engine.
2. Try to install the constructs.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

N/A

### Framework Version

_No response_

### Node.js Version

25.X

### OS

OSX

### Language

Typescript

### Language Version

_No response_

### Region experiencing the issue

N/A

### Code modification

no

### Other information

_No response_

### Service quota

- [x] I have reviewed the service quotas for this construct

--- a/.projenrc.ts
+++ b/.projenrc.ts
@@ -79,7 +79,7 @@ const project = new awscdk.AwsCdkConstructLibrary({
   bundledDeps: ['deepmerge', `@aws-cdk/aws-lambda-python-alpha@${CDK_VERSION}-alpha.0`],
   // Keep synchronized with https://github.com/nodejs/release#release-schedule
   minNodeVersion: '20.x', // 'MAINTENANCE' (first LTS)
-  maxNodeVersion: '24.x', // 'CURRENT'
+  maxNodeVersion: '25.x', // 'CURRENT'
   workflowNodeVersion: '22.x', // 'ACTIVE'
 
   npmTokenSecret: 'NPM_TOKEN',

--- a/package.json
+++ b/package.json
@@ -152,7 +152,7 @@
     "generative-ai"
   ],
   "engines": {
-    "node": ">= 20.x <= 24.x"
+    "node": ">= 20.x <= 25.x"
   },
   "main": "lib/index.js",
   "license": "Apache-2.0",

